gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlanListSerializer.java

```diff
@@ -83,6 +83,14 @@ public JsonElement serialize(List<JobExecutionPlan> src, Type typeOfSrc, JsonSer
       String executionStatus = jobExecutionPlan.getExecutionStatus().name();
       jobExecutionPlanJson.addProperty(SerializationConstants.EXECUTION_STATUS_KEY, executionStatus);
 
+      try {
+        String jobExecutionFuture = jobExecutionPlan.getSpecExecutor().getProducer().get()
+            .serializeAddSpecResponse(jobExecutionPlan.getJobFuture().orNull());
+        jobExecutionPlanJson.addProperty(SerializationConstants.JOB_EXECUTION_FUTURE, jobExecutionFuture);
+      } catch (InterruptedException | ExecutionException e) {
+        log.warn("Error during serialization of JobExecutionFuture.");
```

Comment:
I think I had a comment earlier about this. Not sure, if it got lost. But should we propagate the exception? It may not be much use persisting a dag which cannot be deserialized back.

Reply:
done